internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -135,60 +135,85 @@ func destinationRulesByPort(allPorts []string, destinationRules []*pbauth.Destin
 		for _, p := range allPorts {
 			out[p] = nil
 		}
-
 		return out
 	}
 
 	for _, destinationRule := range destinationRules {
-		ports, dr := convertDestinationRule(allPorts, destinationRule)
-		for _, p := range ports {
-			if dr == nil {
+		portRules := convertDestinationRule(allPorts, destinationRule)
+		for p, pr := range portRules {
+			if pr.rule == nil {
 				out[p] = nil
 				continue
 			}
-			out[p] = append(out[p], dr)
+			out[p] = append(out[p], pr.rule)
 		}
 	}
 
 	return out
 }
 
-//nolint:unparam
-func convertDestinationRule(allPorts []string, dr *pbauth.DestinationRule) ([]string, *pbproxystate.DestinationRule) {
-	ports := make(map[string]struct{})
-	if len(dr.PortNames) > 0 {
-		for _, p := range dr.PortNames {
-			ports[p] = struct{}{}
-		}
-	} else {
-		for _, p := range allPorts {
-			ports[p] = struct{}{}
-		}
-	}
+type PortRule struct {
+	rule *pbproxystate.DestinationRule
```

Comment:
Why are we wrapping the `*pbproxystate.DestinationRule`?

Reply:
because this seemed like a neater solution in order to add an empty placeholder for the case where the destination rule is empty (port only). The alternative I considered was hand implementing the isEmpty functionality on the proto message object, so we could check it on line 144, but that seemed like more extra maintenance than it was worth.